.github/workflows/newfold-prepare-release.yml

```diff
@@ -0,0 +1,40 @@
+name: Newfold Prepare Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      level:
+        description: The level of release to be used.
+        type: choice
+        options:
+          - patch
+          - minor
+          - major
+        default: "patch"
+        required: true
+
+jobs:
+  # This job is used to extract the branch name from the pull request or branch.
+  setup:
+    name: Setup
+    runs-on: ubuntu-latest
+    outputs:
+      branch: ${{ steps.extract_branch.outputs.branch }}
+    steps:
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_OUTPUT"
```

Comment:
Can you explain a bit more why this is necessary? Unless I am missing something, I think that `github.ref_name` or `GITHUB_REF_NAME` should do what we need. If so, this step could be eliminated and `env.GITHUB_REF_NAME` can just be passed to `plugin-branch` in `prep-release`.

`GITHUB_HEAD_REF` is only set for pull requests. Is there ever a situation where we would prepare a release from a pull request?